### ISSUE TITLE
[Feature Request] Implement New Plan Dialog 

### ISSUE THREAD

**Issue by ZzRanger** [UTDNebula/planner#91]
# Overview

The New Plan Dialog is a popup window that appears whenever the button to create new degree plan is clicked. The information collected from this dialog will be used to create the plan metadata. The plan metadata should contain general information about the plan (i.e. major, duration), as well as information that will be used by the degree plan generator.

The dialog has **already been created** and can be found in '/components/planner/NewPlanDialog.tsx'. All that's left is to complete the dialog and integrate it with the rest of the application. 

# Proposal

Here is some information the dialog should get from the user:
- Major/minors
- Degree plan name
- Plan starting date (defaults to current semester)
- Plan duration
- Fast track information
- Social security number
- Average course load (this is a general way for users to specify ~how many courses they want to take per semester)
- Desired graduation date
- Include summer semesters

Feel free to add to this list. Once the user is done, the data collected should be stored in Redux and the external database.

## Supplementary Information

The dialog should somewhat resemble the image below, but it doesn't have to be exactly like it.

![New Plan Creation Flowdialog](https://user-images.githubusercontent.com/47403443/153691267-8ea55209-d6a4-4e68-a563-6c83a173866d.png)

**Comment by lokeshyerneni** [comment]
I will take care of it

**Comment by WillieCubed** [comment]
Just to clarify, the [Figma](https://www.figma.com/file/cycV0bxP6vTJJYUONFoBr0/Planner-Mockups?node-id=26034%3A33623) constitutes the spec for the project, and the implementation should mirror it precisely. However, in this specific instance, knowing that the dialog will be redesigned as part of a revised onboarding and new plan creation flow, it's fine to take some creative liberties with the understanding that it will be cleaned up in the future.

I would also recommend abstracting as much of the dialog into components as possible, namely:
 - The stepper at the bottom
 - The dialog itself
 - The search box functionality for degree plans
 - The selected plans list

**Comment by WillieCubed** [comment]
@rajmeet2001 @ZzRanger Don't forget to add every issue to all relevant projects ([Planner Maintainer Current Tasks](https://github.com/orgs/UTDNebula/projects/6)(https://github.com/orgs/UTDNebula/projects/6) in this case). Additionally, add it to a relevant milestone if you know that this feature is intended to be included in a future release.

**Comment by Shaurya0108** [comment]
I'll take it
